File: pset6/readability/readability.py
from math import *
def letters_num(y):
    #count the letters
    count = 0
    for i in range(len(y)):
        # check if the the char if alphabetical
        if ((y[i] >= 'a'  and y[i] <= 'z') or (y[i] >= 'A'  and y[i] <= 'Z')):
            count += 1
    return count


def word_num(x):
    #count the words
    count = 0
    for i in range(len(x)):
        # check if the the char if space
        if (x[i] == ' '):
            count += 1
    return count + 1


def sent_num(x):
    #count of the sentence
    count = 0
    for i in range(len(x)):
        # check the end of the sentance
        if (x[i] == '.'  or x[i] == '!'  or x[i] == '?'):
            count += 1
    return count


def Avg (x, y):
    #get the average per 100 word
    return x * 100 /  float(y)







# get the text
inp = input("Text: ")
# parameters of the The Coleman-Liau index
L = Avg (letters_num(inp ), word_num(inp))
S = Avg (sent_num(inp),  word_num(inp))
#The Coleman-Liau index formula
G = round(0.0588 * L - 0.296 * S - 15.8)
if (G >= 16):
    print("Grade 16+")
elif (G < 1):
    print("Before Grade 1")
else:
    print("Grade", G)

File: pset2/readability/readability.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <math.h>

int letters_num(string y);
int word_num(string x);
int sent_num(string x);
float avg(int x, int y);

int main(void)
{
    // get the text
    string in = get_string("Text: ");
    // parameters of the The Coleman-Liau index
    float L = avg(letters_num(in), word_num(in));
    float S = avg(sent_num(in),  word_num(in));
    //The Coleman-Liau index formula
    int G = round(0.0588 * L - 0.296 * S - 15.8);
    if (G >= 16)
    {
        printf("Grade 16+\n");
    }
    else if (G < 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade %d\n", G);
    }
}


int letters_num(string y)
{
    //count the letters
    int count = 0;
    for (int i = 0 ; i < strlen(y); i++)
    {
        // check if the the char if alphabetical
        if ((y[i] >= 'a' && y[i] <= 'z') || (y[i] >= 'A' && y[i] <= 'Z'))
        {
            count += 1;
        }
    }
    return count;
}

int word_num(string x)
{
    //count the words
    int count = 0;
    for (int i = 0 ; i < strlen(x); i++)
    {
        // check if the the char if space
        if (x[i] == ' ')
        {
            count += 1;
        }
    }
    return count + 1;
}

int sent_num(string x)
{
    //count of the sentence
    int count = 0;
    for (int i = 0 ; i < strlen(x) ; i++)
    {
        // check the end of the sentance
        if (x[i] == '.' || x[i] == '!' || x[i] == '?')
        {
            count += 1;
        }
    }
    return count;
}

float avg(int x, int y)
{
    //get the average per 100 word
    return x * 100 / (float) y ;
}
